odoo/addons/base/views/ir_actions_views.xml

```diff
@@ -305,7 +305,20 @@
         <menuitem action="ir_client_actions_report" id="menu_ir_client_actions_report" parent="base.next_id_6"/>
 
         <!-- ir.actions.server -->
-
+        <record id="action_record_from_server_action" model="ir.actions.server">
+            <field name="name">View Related Record Of Server Action</field>
+            <field name="model_id" ref="model_ir_actions_server"/>
+            <field name="state">code</field>
+            <field name="code">
+                action = {
```

Comment:
change into one business method on iractionserver
the two other models would call that one
buttons call their respective method